ta/pkcs11/src/pkcs11_attributes.c

```diff
@@ -1660,3 +1761,70 @@ enum pkcs11_rc set_key_data(struct obj_attrs **head, void *data,
 		return PKCS11_CKR_GENERAL_ERROR;
 	}
 }
+
+enum pkcs11_rc add_missing_attribute_id(struct obj_attrs **attrs1,
```

Comment:
I ported and fixed some cases in the code here... but when I was checking it out against I don't think we are supposed to generate value for CKA_ID. Left it here so that we can discuss it.

Reply:
As an example when someone uploads certificate then it is set to match the same CKA_ID as matching public key (and private key).

But for me it seems that it is job of user space to set CKA_ID.

How it should be handled in key pair gen template -- should we just carry what is provided for particular key or should there be any copying of the CKA_ID from public template -> private key or vise versa.

Reply:
In `C_GenerateKeyPair` example in specification CKA_ID is only in private template and there is indication that they should be same.

So perhaps the best action would be to cross copy them should one have it defined -- otherwise leave it as empty.

Reply:
Updated the PR about this.

Reply:
this comment thread is resolved, i think.